src/Sylius/Component/Promotion/Checker/CouponsEligibilityChecker.php

```diff
@@ -46,8 +46,6 @@ public function isEligible(PromotionSubjectInterface $subject, PromotionInterfac
         }
 
         if (!$this->isCouponEligible($promotion, $subject)) {
-            $this->eventDispatcher->dispatch(SyliusPromotionEvents::COUPON_NOT_ELIGIBLE, new GenericEvent($promotion));
```

Comment:
I think this could be used to issue messages like: in order to use this coupon, you need to comply with these rules: 
- have O(n) product(s) from taxon ZXY
- cart total should be greater than

ofc, this is something I need in my project, most probably as a userLand implementation.